### ISSUE TITLE
Make experiments with layers

### ISSUE THREAD

**Issue by kris701** [jamadaha/P5#80]

**Comment by kris701** [comment]
This is in relation to #81

**Comment by kris701** [comment]
Note for a potential experiment (From source [article](https://arxiv.org/abs/1511.06434))
```
Architecture guidelines for stable Deep Convolutional GANs
  • Replace any pooling layers with strided convolutions (discriminator) and fractional-strided convolutions (generator).
  • Use batchnorm in both the generator and the discriminator.
  • Remove fully connected hidden layers for deeper architectures.
  • Use ReLU activation in generator for all layers except for the output, which uses Tanh.
  • Use LeakyReLU activation in the discriminator for all layers.

**Comment by kris701** [comment]
List of things to consider when training a [CGAN](https://machinelearningmastery.com/how-to-train-stable-generative-adversarial-networks/):
```
A summary of some of the more actionable tips is provided below.
* Normalize inputs to the range [-1, 1] and use tanh in the generator output.
* Flip the labels and loss function when training the generator.
* Sample Gaussian random numbers as input to the generator.
* Use mini batches of all real or all fake for calculating batch norm statistics.
* Use Leaky ReLU in the generator and discriminator.
* Use Average pooling and stride for downsampling; use ConvTranspose2D and stride for upsampling.
* Use label smoothing in the discriminator, with small random noise.
* Add random noise to the labels in the discriminator.
* Use DCGAN architecture, unless you have a good reason not to.
* A loss of 0.0 in the discriminator is a failure mode.
* If loss of the generator steadily decreases, it is likely fooling the discriminator with garbage images.
* Use labels if you have them.
* Add noise to inputs to the discriminator and decay the noise over time.
* Use dropout of 50 percent during train and generation.
